libraries/standard/http/src/http_client.c

```diff
@@ -333,15 +557,31 @@ HTTPStatus_t HTTPClient_Send( const HTTPTransportInterface_t * pTransport,
     /* Send the body, which is at another location in memory. */
     if( returnStatus == HTTP_SUCCESS )
     {
-        returnStatus = _sendHttpBody( pTransport,
-                                      pRequestBodyBuf,
-                                      reqBodyBufLen );
+        if( pRequestBodyBuf != NULL )
+        {
+            returnStatus = _sendHttpBody( pTransport,
+                                          pRequestBodyBuf,
+                                          reqBodyBufLen );
+        }
+        else
+        {
+            IotLogDebug( "A request body was not sent: pRequestBodyBuf is NULL." );
+        }
     }
 
     if( returnStatus == HTTP_SUCCESS )
     {
-        returnStatus = _receiveHttpResponse( pTransport,
-                                             pResponse );
+        /* If the application chooses to receive a response, then pResponse
+         * will not be NULL. */
+        if( pResponse != NULL )
+        {
+            returnStatus = _receiveAndParseHttpResponse( pTransport,
+                                                         pResponse );
+        }
+        else
+        {
+            IotLogWarn( "A response was not received: pResponse is NULL. " );
```

Comment:
If `pResponse` is optional, then the message should not be logged as a warning, because the API considers it _fine_ to not provide the response buffer.

Reply:
Also, the message is misleading as the library did not attempt to read from the transport layer